src/System.CommandLine.Tests/CustomParsingTests.cs

```diff
@@ -0,0 +1,910 @@
+// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Collections.Generic;
+using System.CommandLine.Parsing;
+using System.CommandLine.Tests.Utility;
+using System.IO;
+using System.Linq;
+using System.Net;
+using System.Threading.Tasks;
+using FluentAssertions;
+using Xunit;
+
+namespace System.CommandLine.Tests;
+
+public class CustomParsingTests
```

Comment:
That's correct. 

The cleanups and this are in separate commits to make reviewing simpler. I moved this out to a separate class because not all of the custom parsing tests are argument tests, and because I'll be adding more tests here to cover using name-based lookups within custom parsers.